server/bin/www

```diff
@@ -33,17 +37,47 @@ var server = http.createServer(app);
 //server.listen(port);
 //server.on("error", onError);
 //server.on("listening", onListening);
-models.sequelize.sync().then(function () {
+models.sequelize.sync().then(function() {
   /**
    * Listen on provided port, on all network interfaces.
    */
-  server.listen(port, function () {
+  server.listen(port, function() {
     debug("Express server listening on port " + server.address().port);
   });
   server.on("error", onError);
   server.on("listening", onListening);
 });
 
+/**
+ * Socket setup
+ */
+const io = socketio(server);
+
+io.on("connection", socket => {
+  console.log("We have a new connection.");
+  let encodedCookie = socket.handshake.headers.cookie;
+  var parsedCookies = cookie.parse(encodedCookie);
+
+  // User data of current Socket User stored in userData
+  // User data includes: userName, userEmail and userId
+  var userData = authJWTCheck(parsedCookies);
+
```

Comment:
if the JWT is invalid, drop the connection ( emmit "disconnect")
else, add to the clients object: `socketClients[userData.userId] = socket.id`